src/main/java/uk/gov/ons/census/action/schedule/FulfilmentProcessor.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.ons.census.action.schedule;
+
+import java.util.List;
+import java.util.UUID;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.action.model.dto.FulfilmentMapperDTO;
+
+@Component
+public class FulfilmentProcessor {
+
+  private JdbcTemplate jdbcTemplate;
+
+  public FulfilmentProcessor(JdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  public void addFulfilmentBatchIdAndQuantity() {
+    String query =
+        "select fulfilment_code, count(*) from actionv2.fulfilments_to_send group by fulfilment_code";
+    List<FulfilmentMapperDTO> fulfilmentsToSend =
+        jdbcTemplate.query(query, new uk.gov.ons.census.action.utility.FulfilmentMapper());
+
+    fulfilmentsToSend.forEach(
+        (fulfilment) -> {
+          jdbcTemplate.update(
```

Comment:
We shouldn't use hand-written SQL unless we're doing something which JPA isn't capable of doing.

This could be re-written as simply:
```
fulfilment.setCount(x);
fulfilment.setBatchId(x);
fulfilmentRepository.save();
```